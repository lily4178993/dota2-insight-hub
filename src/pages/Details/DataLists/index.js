export { default as PromatchesList } from './PromatchesList';
export { default as ProPlayersList } from './ProPlayersList';
export { default as ItemsList } from './ItemsList';
export { default as HeroesList } from './HeroesList';
