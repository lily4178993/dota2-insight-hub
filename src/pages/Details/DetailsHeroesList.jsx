import React from 'react'

const DetailsHeroesList = () => {
  // Will be implemented in a future version
  // For now, it will just return a placeholder     
  // This component will be used to display the list of heroes
  return (
    <div>DetailsHeroesList</div>
  )
}

export default DetailsHeroesList