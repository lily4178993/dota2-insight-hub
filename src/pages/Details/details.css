.detailsList-container {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.detailsList-container .chess-grid {
  width: 25%;
  color: var(--color-white);
}

.detailsList-container .chess-grid:nth-child(8n + 1),
.detailsList-container .chess-grid:nth-child(8n + 3),
.detailsList-container .chess-grid:nth-child(8n + 6),
.detailsList-container .chess-grid:nth-child(8n + 8) {
  background: var(--color-blue-2);
}

.detailsList-container .chess-grid:nth-child(8n + 2),
.detailsList-container .chess-grid:nth-child(8n + 4),
.detailsList-container .chess-grid:nth-child(8n + 5),
.detailsList-container .chess-grid:nth-child(8n + 7) {
  background: var(--color-blue-3);
}

@media (max-width: 768px) {
  .detailsList-container .chess-grid {
    width: 50%;
  }

  .detailsList-container .chess-grid:nth-child(8n + 1),
  .detailsList-container .chess-grid:nth-child(8n + 4),
  .detailsList-container .chess-grid:nth-child(8n + 5),
  .detailsList-container .chess-grid:nth-child(8n + 8) {
    background: var(--color-blue-2);
  }

  .detailsList-container .chess-grid:nth-child(8n + 2),
  .detailsList-container .chess-grid:nth-child(8n + 3),
  .detailsList-container .chess-grid:nth-child(8n + 6),
  .detailsList-container .chess-grid:nth-child(8n + 7) {
    background: var(--color-blue-3);
  }
}
