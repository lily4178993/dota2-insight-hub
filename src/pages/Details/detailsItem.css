.error-fetch {
  font-size: 1.4em;
  text-align: center;
  margin: 1em auto;
  padding: 1em;
}
