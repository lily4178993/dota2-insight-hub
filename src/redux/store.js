import { configureStore/* , getDefaultMiddleware */ } from '@reduxjs/toolkit';
/* import logger from 'redux-logger'; */
import heroReducer from './slices/heroesSlice';
import itemReducer from './slices/itemsSlice';
import matchReducer from './slices/matchesSlice';
import playerReducer from './slices/playersSlice';

const store = configureStore({
  reducer: {
    heroes: heroReducer,
    items: itemReducer,
    matches: matchReducer,
    players: playerReducer,
  },
  /* middleware: [...getDefaultMiddleware(), logger], */
});

export default store;
