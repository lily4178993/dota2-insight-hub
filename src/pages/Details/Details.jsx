import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { Card, Filter } from '../../components';
import matchesImage from '../../assets/trophy.png';
import './details.css';

function Details() {
  const { detailsListName } = useParams();
  const detailsData = useSelector((state) => {
    switch (detailsListName) {
      case 'heroes':
        return state.heroes.heroes;
      case 'items':
        return state.items.items;
      case 'proMatches':
        return state.matches.matches;
      case 'proPlayers':
        return state.players.players;
      default:
        return null;
    }
  });

  // Ensure that detailsData is an array
  const detailsDataArray = Array.isArray(detailsData) ? detailsData : [];

  const mapDataProperties = (data) => {
    switch (detailsListName) {
      case 'heroes':
        return {
          cardImage: `https://api.opendota.com${data?.img}`,
          cardTitle: data?.localized_name,
          cardCount: data?.base_str,
          isParser: true,
          dataKey: data?.hero_id,
        };
      case 'items':
        return {
          cardImage: `https://api.opendota.com${data?.img}`,
          cardTitle: data?.dname,
          cardCount: data?.cost,
          dataKey: data?.key,
        };
      case 'proMatches':
        return {
          cardImage: matchesImage,
          cardTitle: data?.dire_name,
          cardCount: data?.dire_score,
          dataKey: data?.match_id,
        };
      case 'proPlayers':
        return {
          cardImage: data?.avatarmedium,
          cardTitle: data?.personaname,
          cardCount: data?.account_id,
          dataKey: data?.account_id,
        };
      default:
        return null;
    }
  };

  const [filter, setFilter] = useState('Names');

  // Define a function to filter and sort cards based on the chosen filter
  const filteredAndSortedCards = () => {
    // Filter cards based on the filter type
    let filteredCards = [...detailsDataArray].map(mapDataProperties);

    if (filter === 'Counts') {
      // Sort the cards by Count in descending order
      filteredCards = filteredCards.sort((a, b) => b.cardCount - a.cardCount);
    }

    return filteredCards;
  };

  return (
    <section>
      <Filter setFilter={setFilter} />
      <div className="detailsList-container">
        {detailsDataArray.length > 0 ? (
          filteredAndSortedCards().map((data) => {
            if (!data) {
              return null;
            }

            return (
              <Link to={`/details/${detailsListName}/${data?.dataKey}`} key={data?.dataKey}>
                <Card
                  cardImage={data?.cardImage}
                  cardTitle={data?.cardTitle}
                  cardCount={data?.cardCount}
                  isParser={data?.isParser}
                />
              </Link>
            );
          })
        ) : (
          'Data not found'
        )}
      </div>
    </section>
  );
}
export default Details;
