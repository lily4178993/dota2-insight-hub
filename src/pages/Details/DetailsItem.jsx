import React from 'react';
import { useParams, useLocation } from 'react-router-dom';
import { useSelector } from 'react-redux';
import {
  HeroInfo,
  ItemPresentation,
  MatchInfo,
  ProPlayerProfile,
} from '../../components';
import './detailsItem.css';

const DetailsItem = () => {
  const { detailsItemID } = useParams();
  const location = useLocation();
  const detailsListName = location.pathname.split('/')[2];
  const detailsData = useSelector((state) => {
    switch (detailsListName) {
      case 'heroes':
        return state.heroes.heroes;
      case 'items':
        return state.items.items;
      case 'proMatches':
        return state.matches.matches;
      case 'proPlayers':
        return state.players.players;
      default:
        return [];
    }
  });

  const renderComponent = () => {
    switch (detailsListName) {
      case 'heroes':
        return detailsData[detailsItemID] ? (
          <HeroInfo data={detailsData[detailsItemID - 1]} />
        ) : (
          <div className="error-fetch">
            Something went wrong!
            <br />
            Hero&quot;s data not found
          </div>
        );
      case 'items':
        // eslint-disable-next-line no-case-declarations
        const selectedItem = detailsData.find((item) => item.key === detailsItemID);
        return selectedItem ? (
          <ItemPresentation data={selectedItem} />
        ) : (
          <div className="error-fetch">
            Something went wrong!
            <br />
            Item&quot;s data not found
          </div>
        );
      case 'proMatches':
        // eslint-disable-next-line no-case-declarations
        const index = detailsData.findIndex((object) => object.match_id === Number(detailsItemID));
        if (index !== -1) return <MatchInfo data={detailsData[index]} />;
        return (
          <div className="error-fetch">
            Something went wrong!
            <br />
            Match&quot;s data not found
          </div>
        );
      case 'proPlayers':
        // eslint-disable-next-line no-case-declarations
        const index2 = detailsData.findIndex(
          (object) => object.account_id === Number(detailsItemID),
        );
        if (index2 !== -1) return <ProPlayerProfile data={detailsData[index2]} />;
        return (
          <div className="error-fetch">
            Something went wrong!
            <br />
            Player&quot;s data not found
          </div>
        );
      default:
        return (
          <div className="error-fetch">
            Something went wrong!
            <br />
            Details not found
          </div>
        );
    }
  };

  return <section>{renderComponent()}</section>;
};

export default DetailsItem;
