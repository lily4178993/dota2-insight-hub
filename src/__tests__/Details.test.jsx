import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Details from '../pages/Details/Details';
import store from '../redux/store';

test('Details component snapshot', () => {
  const { asFragment } = render(
    <Router>
      <Provider store={store}>
        <Details />
      </Provider>
    </Router>,
  );

  expect(asFragment()).toMatchSnapshot();
});
