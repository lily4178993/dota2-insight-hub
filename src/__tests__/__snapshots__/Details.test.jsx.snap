// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="header-filter-box"
    >
      <span>
        Choose a Card
      </span>
      <button
        title="Filter"
        type="button"
      >
        Filter 
        <i
          class="bx bx-down-arrow-circle"
          style="transform: rotate(0deg);"
        />
      </button>
      <div
        class="filter-box"
        style="right: -100%;"
      >
        <ul>
          <li>
            <button
              title="Filter by Names"
              type="button"
            >
              Names
            </button>
          </li>
          <li>
            <button
              title="Filter by Counts"
              type="button"
            >
              Counts
            </button>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="detailsList-container"
    >
      Data not found
    </div>
  </section>
</DocumentFragment>
`;
